encoding/encoding.go

```diff
@@ -51,6 +52,20 @@ type Serializer interface {
 	CheckSuccess(body *transport.Response) error
 }
 
+// StreamRequestReader interface exposes method to read multiple request body
+type StreamRequestReader interface {
+	// NextBody returns the encoded request body if available, and if not, returns an
+	// io.EOF error
+	NextBody() ([]byte, error)
+}
+
+// StreamSerializer interface exposes additional methods to handle stream requests
```

Comment:
for consistency, the current `Serializer` interface comment reads as:
`Serializer serializes and deserializes data for a specific encoding and method.`
You should probably do `StreamSerializer serializes and deserializes data for a stream request` ?